build/__functions.sh

```diff
@@ -78,3 +78,23 @@ function showDockerImageSizes()
 {
 	docker system df -v
 }
+
+function shouldStageRuntimeVersion()
+{
+	platformName="$1"
+	platformRuntimeVersion="$2"
+
+	source $REPO_DIR/build/__stagingRunTimeConstants.sh
+
+	case $platformName in
+	'dotnet'|'dotnetcore')
+		if [[ " ${DOTNETCORE_STAGING_RUNTIME_VERSIONS[*]} " =~ " ${platformRuntimeVersion} " ]]; then
+			return 0
+		fi
+		;;
+	*) 
+		echo "Platform '$platformName' does not support staging."
```

Comment:
I'm mostly curious here: why should a platform "support" staging? What is required for this, and that make it non trivial?
The answer might be in the rest of the PR, so I might delete the comment later.

Reply:
It will just need a few things to be added to "support" staging, such as a list constant to the constants.yaml, such as what dotnet has with `dotnetcore-staging-runtime-versions`. There will also need to be updates to their platform specific generateDockerfiles scripts https://github.com/microsoft/Oryx/pull/1643/files#diff-db172e07e3c1672513a6c8ce0c9248f8522602a54038c6459aefb8af98946a39 and probably the build runtime bases script https://github.com/microsoft/Oryx/pull/1643/files#diff-bccc0a4c8d28421a96eefefb692ff896fabcb915f0b2ae9e31cceb9b04de0e48 

So not a ton of work to make all platforms support staging, but non-trivial enough that it seemed out of scope for the immediate effort?

Reply:
Makes sense. We can probably keep this as a future effort once we merge staging with the dev environment maybe (until then, there wouldn't be a lot of interests in expanding this outside of .NET anyway).